enhancers_pod.go

```diff
@@ -81,14 +81,20 @@ func runPodEnhancer(ctx context.Context, podMeta *v1.ObjectReference, cachedObje
 		sentryEvent.Fingerprint = []string{message}
 	}
 
-	if len(pod.OwnerReferences) > 0 {
-		// If the pod is controlled by something (e.g. a replicaset), group all issues
-		// for all controlled pod together.
-		owner := pod.OwnerReferences[0]
-		sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, owner.Kind, owner.Name)
-	} else {
+	// Using finger print to group events together
+	// The pod is not owned by a higher resource
+	if len(pod.OwnerReferences) == 0 {
 		// Standalone pod => most probably it has a unique name
 		sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, podName)
+		// The pod is owned by a higher resource
+	} else {
+		// The pod is owned by a cronJob (as grandchild workload resource)
```

Comment:
Since `len(pod.OwnerReferences) > 0` doesn't necessarily mean that the pod is owned by a cronjob: let's check `pod.OwnerReferences[0].Kind` (let's ignore cases when there's more than one owner for now), and that the pod is owned by a Job.
If the pod is owned by a job -- it _might_ be owned by a CronJob, so we can try to do what you do in `enhancers_crons` now. 

Basically, let's attempt `runCronsEnhancer` only when we at least know that the direct owner is a Job.

Reply:
sounds good